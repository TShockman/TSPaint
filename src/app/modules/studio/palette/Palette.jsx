import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import ColorCell from '../colorCell';
import theme from './palette.scss';

export default class Palette extends PureComponent {
  static propTypes = {
    selectColor: PropTypes.func.isRequired,
    selected: PropTypes.string.isRequired,
    colors: PropTypes.arrayOf(PropTypes.string).isRequired
  };

  render() {
    const {selectColor, selected, colors} = this.props;
    return (
        <div className="palette">
          {colors.map((color, i) => {
            const active = color === selected ? "active" : "";
            return <ColorCell key={i} className={active} index={i} color={color} onSelect={selectColor}/>;
          })}
        </div>
    )
  }
}