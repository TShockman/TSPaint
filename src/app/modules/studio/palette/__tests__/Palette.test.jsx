import Palette from '../index';
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ReactShallowRenderer();

describe('Palette', () => {
  it('renders consistently, marking active color', () => {
    const props = {
      selectColor: jest.fn(),
      selected: "#555555",
      colors: ["#000000", "#333333", "#555555", "#FFFFFF"]
    };
    const palette = <Palette {...props}/>
    expect(renderer.render(palette)).toMatchSnapshot();
  });
});