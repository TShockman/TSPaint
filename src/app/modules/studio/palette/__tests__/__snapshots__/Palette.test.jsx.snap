// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Palette renders consistently, marking active color 1`] = `
<div
  className="palette"
>
  <ColorCell
    className=""
    color="#000000"
    index={0}
    onSelect={[MockFunction]}
  />
  <ColorCell
    className=""
    color="#333333"
    index={1}
    onSelect={[MockFunction]}
  />
  <ColorCell
    className="active"
    color="#555555"
    index={2}
    onSelect={[MockFunction]}
  />
  <ColorCell
    className=""
    color="#FFFFFF"
    index={3}
    onSelect={[MockFunction]}
  />
</div>
`;
